designdocs/passkey_p256.md

```diff
@@ -0,0 +1,552 @@
+
+        let passkeyCall = api.registry.createType('PalletFrequencyTxPaymentPasskeyCall', passkeyCallData);
+    ```
+
+2. **Passkey Signing**:
+   - The transaction payload is sent to the Passkey for signing.
+   - The Passkey signs the transaction payload using its private key, producing a `passkey_tx_signature`.
+
+    ```javascript
+
+      // Sign the passkeyCall payload using Passkey
+      const passkeyTxSignature = await SimpleWebAuthnBrowser.startAssertion({
+        challenge: passkeyCall.toU8a(),
+        // Additional options
+      });
+
+      // The signature generated by Passkey
+      passkey_tx_signature = passkeyTxSignature.signature;
+      passkey_authenticator_data = passkeyTxSignature.authenticatorData;
+      passkey_client_data = passkeyTxSignature.clientData;
+      //
+    ```
+
+3. **Transaction Submission**:
+   - The transaction payload, that requires, `passkey_pk`, `passkey_tx_signature`, `passkey_authenticator_data`, `passkey_client_data` is submitted to the chain.
+   - The transaction is submitted to the chain as in `unsigned` extrinsic.
+
+    ```javascript
+      // Prepare the payload for the passkey proxy call
+      let payload = {
+        passkeyPublicKey: Array.from(passkey_pk),
+        passkeySignature: Array.from(base64UrlToUint8Array(passkey_tx_signature)),
+        passkeyAuthenticator: Array.from(base64UrlToUint8Array(passkey_authenticator_data)),
+        passkeyClientDataJson: Array.from(base64UrlToUint8Array(
+          passkey_client_data)),
+        passkeyCall,
+      };
+
+      const passkey_payload = api.createType('PalletFrequencyTxPaymentPasskeyPayload', payload);
+
+      // Submit the transaction to the chain without signature
+      const tx = api.tx.{frequecyPassKeyProxy}.submitPasskeyTx(passkey_payload);
+      await tx.send()
+    ```
+
+4. **Backend Validation and Submission**:
+    - The backend can further validate the transaction payload before submitting it to the chain.
+    - The backend validates the `passkey_tx_signature` and the `passkey_authenticator_data` to ensure the authenticity of the transaction.
+    - For more secure operations, handling the transaction creation and submission on client side is recommended.
+
+### Security Considerations
+
+#### Generic
+
+- Never store or backup the seed phrase other than memory.
+- Prefer more client side data processing and handling, backend should only store the necessary data.
+- Choose vetted libraries, npm packages for handling cryptographic operations, such that signing with seed phrase, or handling of private keys is done securely.
+
+#### Front-end (client)
+
+- If key generation is done in front-end, it should ideally being done inside an isolated section such as iframe or Web Worker.
+- Generated Keypair should not get stored permanently (except for back up options) and removed as soon as it is not required.
+
+#### Backend
+
+- Passkey registration response should get verified which checks the random challenge
```

Comment:
This is the first time a random challenge is mentioned.  What is the random challenge here?

Reply:
Added details